api/environments/identities/models.py

```diff
@@ -163,7 +163,9 @@ def get_segments(
         :return: List of matching segments
         """
         matching_segments = []
-        traits = self.identity_traits.all() if traits is None else traits
+        traits = (
+            self.identity_traits.all() if (traits is None and self.id) else traits or []
```

Comment:
I'm curious to know which scenarios exist where we don't have `self.id` set and we're calling this method?

Reply:
For transient identity, test in question: `test_get_identities__transient__no_persistence`